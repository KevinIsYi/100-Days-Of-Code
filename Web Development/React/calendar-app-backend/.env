PORT=4000
DB_CONNECTION=mongodb+srv://mern_user:NekxHA9lixWzsT0c@cluster0.9xvmk.mongodb.net/mern_calendar
JWT_SEED=p@L@BR@_S3CR3T@_JWT