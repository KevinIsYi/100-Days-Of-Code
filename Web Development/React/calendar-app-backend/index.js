const express = require('express');
const dotenv = require('dotenv');
const cors = require('cors');
const { dbConnection } = require('./database/config');

dotenv.config();

// Create express server
const app = express();

// Database
dbConnection();


//CORS
app.use(cors());

app.use(express.static('public'));

// Lecture and parse of body
app.use( express.json() );

app.use('/api/auth', require('./routes/auth'));
app.use('/api/events', require('./routes/events'));


// Listen petitions
app.listen(process.env.PORT, () => { // Port, callback
    console.log(`Server is running on: ${ process.env.PORT }`);
});