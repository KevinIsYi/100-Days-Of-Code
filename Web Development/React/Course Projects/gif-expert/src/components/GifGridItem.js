import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({url}) => {
    return (
        <img className="animate__fadeInLeft" src={url} alt="gif">
        </img>
    )
}

GifGridItem.propTypes = {
    url: PropTypes.string.isRequired
}
