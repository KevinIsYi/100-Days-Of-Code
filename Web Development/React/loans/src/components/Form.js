import React, { useState } from 'react';
import { calculateTotal } from '../helpers/helpers';

export const Form = ({ quantity, setQuantity, term, setTerm, setTotal, setIsLoading }) => {

    const [ error, setError ] = useState(false);    

    const handleChange = (e) => {
        setQuantity(e.target.value);
    }

    const handleSelectChange = (e) => {
        setTerm(e.target.value);
    }

    const calculateLoan = (e) => {
        e.preventDefault();
        if (term === '' || quantity === 0) {
            setError(true);
        }
        else {
            setIsLoading(true);
            setTimeout(() => {
                setError(false);
                const total = calculateTotal(quantity, term);
                setTotal(total);
                setIsLoading(false);
            }, 1000);
        }
    }

    return (
        <>
            <form onSubmit={ calculateLoan }>
            <div className="row">
                <div>
                    <label>Cantidad Prestamo</label>
                    <input 
                        className="u-full-width" 
                        type="number" 
                        placeholder="Ejemplo: 3000"
                        value={ quantity }
                        onChange={ handleChange }
                    />
                </div>
                <div>
                    <label>Plazo para Pagar</label>
                    <select 
                        className="u-full-width"
                        onChange={ handleSelectChange }
                        value={ term }
                    >
                        <option value="">Seleccionar</option>
                        <option value="3">3 meses</option>
                        <option value="6">6 meses</option>
                        <option value="12">12 meses</option>
                        <option value="24">24 meses</option>
                    </select>
                </div>
                <div>
                    <input 
                        type="submit" 
                        value="Calcular" 
                        className="button-primary u-full-width" 
                    />
                </div>
            </div>
            </form>
            {
                error && <p className="error">All fields are obligatory</p> 
            }
        </>
    )
}
