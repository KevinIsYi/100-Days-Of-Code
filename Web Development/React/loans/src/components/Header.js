import React from 'react'

export const Header = ({ tittle }) => <h1>{ tittle }</h1>
