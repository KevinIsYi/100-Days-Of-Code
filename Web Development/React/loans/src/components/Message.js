import React from 'react'

export const Message = () => <p>Add a Quantity and Term</p>
