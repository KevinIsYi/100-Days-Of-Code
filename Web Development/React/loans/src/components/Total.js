import React from 'react'

export const Total = ({ total, term, quantity }) => (
        <div className="u-full-width resultado">
            <h2>Resumen</h2>
            <p>Quantity: ${ quantity }</p>
            <p>To pay in: { term } months</p>
            <p>Month Payment: ${ (total / term).toFixed(2) }</p>
            <p>Total: ${ total.toFixed(2) }</p>
        </div>
)
