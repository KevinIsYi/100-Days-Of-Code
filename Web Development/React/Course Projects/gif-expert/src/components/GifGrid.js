import React /*, { useEffect, useState }*/ from 'react'
//import { getGif } from "../helpers/getGifs"; 
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';


export const GifGrid = ({ category }) => {

    const {data, loading} = useFetchGifs(category);

    // const [images, setImages] = useState([]);

    // useEffect(() => {
    //     getGif(category)
    //         .then(imgs => setImages(imgs))
    // }, [category]); // This instruction will be executed when content is rended the first time

    return (
        <>
            <h3>{category}</h3>
            <ul>
                {
                    data.map(({id, url}) => 
                        <li key={id}>  
                            <GifGridItem url={url} />
                        </li>
                    )
                }
            </ul>
        </>
    )
}
