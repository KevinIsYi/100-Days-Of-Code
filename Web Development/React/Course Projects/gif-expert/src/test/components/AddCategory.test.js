import React from 'react';
import {AddCategory} from "../../components/AddCategory";
const { shallow } = require("enzyme")

describe ("Test on <AddCategory", () => {

    const setCategory = () => {};
    const wrapper = shallow(<AddCategory setCategories={setCategory}/>);
    
    test ("Must show correctly", () => {

        expect(wrapper).toMatchSnapshot();
    });
    test("text in textbox must change", () => {
        const input = wrapper.find("input");
        const value = "Hello World!";

        input.simulate("change", {
            target: {
                value: value
        }});

        expect(wrapper.find("p").text().trim()).toBe(value);
    })
});