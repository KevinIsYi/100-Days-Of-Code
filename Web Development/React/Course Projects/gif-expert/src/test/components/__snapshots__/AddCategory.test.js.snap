// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <AddCategory Must show correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
